tinygrad/runtime/ops_python.py

```diff
@@ -46,6 +46,22 @@ def _store(m, i, v):
   if i<0 or i>=len(m): raise IndexError(f"store out of bounds, size is {len(m)}, access is {i}, value is {v}")
   m[i] = v
 
+def cstyle_cast(value, in_dtype: DType, out_dtype: DType, bitcast: bool):
+  def c_bitcast(value, in_dtype: DType, out_dtype: DType):
+    # mapping dtype.name to ctypes API names as per https://docs.python.org/3/library/ctypes.html
+    def to_cname(dtype_name: str): return dtype_name.replace("char", "int8").replace("unsigned", "u").replace(" ", "")
+    in_cvalue, out_ctype = getattr(ctypes, f"c_{to_cname(in_dtype.name)}")(value), getattr(ctypes, f"c_{to_cname(out_dtype.name)}")
+    return ctypes.cast(ctypes.pointer(in_cvalue), ctypes.POINTER(out_ctype)).contents.value
+
+  if bitcast: return c_bitcast(value, in_dtype, out_dtype)
+
+  # perform Python type casting
+  out = int(value) if dtypes.is_int(out_dtype) else float(value) if dtypes.is_float(out_dtype) else bool(value)
+  # return the output if the involved types are Python-supported
+  if not (dtypes.is_unsigned(in_dtype) or dtypes.is_unsigned(out_dtype)): return out
+  # otherwise, bitcast the output to mimic C-style type casting for operands involving unsigned types
+  return c_bitcast(out, out_dtype, out_dtype)
```

Comment:
i think it's more readable to write
```
if dtypes.is_unsigned(in_dtype) or dtypes.is_unsigned(out_dtype): out = c_bitcast(out, out_dtype, out_dtype)
return out
```

also it's not obvious to me why it's `c_bitcast(out, out_dtype, out_dtype)` not `c_bitcast(out, in_dtype, out_dtype)`, consider commenting if that's intended

Reply:
It's because after the Python-cast, they are now both of the same type, and you need to specify the appropriate type to wrap the value correctly in C.

Reply:
A comment feels a bit unnecessary and redundant.

Reply:
I could rewrite the `in_dtype` as `out_dtype` on the same casting line if it feels cleaner.

Reply:
I might have a refactor in mind that makes this even cleaner.

Reply:
I pushed a refactor that addresses this by changing the API of `c_bitcast` to accommodate the use case here, and I explained the reasoning in a comment.